src/lib/dexrt.cpp

```diff
@@ -380,6 +380,16 @@ void dex_ensure_has_cuda_context() {
   }
 }
 
+void dex_get_cuda_architecture(int device, char* arch) {
+  int majorVersion, minorVersion;
+  cuInit(0);
+  CUdevice dev;
+  CHECK(cuDeviceGet, &dev, device);
+  cuDeviceGetAttribute(&majorVersion, CU_DEVICE_ATTRIBUTE_COMPUTE_CAPABILITY_MAJOR, device);
+  cuDeviceGetAttribute(&minorVersion, CU_DEVICE_ATTRIBUTE_COMPUTE_CAPABILITY_MINOR, device);
+  sprintf(arch, "sm_%d%d", majorVersion, minorVersion);
```

Comment:
Can you please use `snprintf` to ensure there will be no overflows? Also good to check that both major and minor are between 0 and 9.

Reply:
Done!